DataFormats/L1Trigger/interface/Vertex.h

```diff
@@ -0,0 +1,34 @@
+#ifndef DataFormats_L1TVertex_Vertex_h
+#define DataFormats_L1TVertex_Vertex_h
+
+#include <vector>
+
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+
+namespace l1t {
+
+  class Vertex;
+  typedef std::vector<Vertex> VertexCollection;
+
+  class Vertex {
+  public:
+    typedef TTTrack<Ref_Phase2TrackerDigi_> Track_t;
+
+    Vertex();
+    Vertex(float z0, const std::vector<edm::Ptr<Track_t>>& tracks);
+    ~Vertex();
+
+    float z0() const;
+
+    const std::vector<edm::Ptr<Track_t>>& tracks() const;
+
+  private:
+    float z0_;
+    std::vector<edm::Ptr<Track_t>> tracks_;
```

Comment:
Just to double-check, can the `edm::Ptr<Track_t>` point to `Track_t` objects in different Event products? (if not, `edm::PtrVector<Track_t>` would be more efficient)

Reply:
No, all of the `Track_t` objects should be from the same event.

Reply:
I meant two (or more) collections in the same event

Reply:
It's possible. This doesn't happen right now, but I can come up with a contrived situation where you might be joining two `Track_t` collections and then forming a vertex (i.e. prompt and displaces). Unless we want to say that these sorts of contrived situations are not acceptable I would recommend leaving the code the way it is.